cpp/include/cugraph_c/centrality_algorithms.h

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cugraph_c/array.h>
+#include <cugraph_c/error.h>
+#include <cugraph_c/graph.h>
+#include <cugraph_c/resource_handle.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief     Opaque centrality result type
+ */
+typedef struct {
+  int32_t align_;
+} cugraph_centrality_result_t;
+
+/**
+ * @brief     Get the vertex ids from the centrality result
+ *
+ * @param [in]   result   The result from a centrality algorithm
+ * @return type erased array of vertex ids
+ */
+cugraph_type_erased_device_array_view_t* cugraph_centrality_result_get_vertices(
+  cugraph_centrality_result_t* result);
+
+/**
+ * @brief     Get the centrality values from a centrality algorithm result
+ *
+ * @param [in]   result   The result from a centrality algorithm
+ * @return type erased array view of centrality values
+ */
+cugraph_type_erased_device_array_view_t* cugraph_centrality_result_get_values(
+  cugraph_centrality_result_t* result);
```

Comment:
Isn't this an update happened in one of the previous PRs? It seems like this PR includes updates for PageRank, HITS, and so on in addition to the updates for Katz. Is this because this PR is showing updates from other PRs or these are indeed new updates in this PR?

Reply:
I felt like the `algorithms.h` file was getting too large to keep track of.  This PR (as well as #2185) starts moving like algorithms into a separate file.  In this case I'm moving all of the centrality based algorithms into `centrality_algorithms.h`.

So the definition of this function was in an earlier PR, this is just moving it to a new file.

Reply:
Gotcha, makes sense. Not for this PR, but to be consistent, PageRank is currently under `link_analysis`. Maybe in a separate PR, shouldn't we better move PageRank to the `centrality` directory? (in C/C++/Python).

Reply:
And HITS as well (https://github.com/rapidsai/cugraph/tree/branch-22.06/cpp/src/link_analysis)

Reply:
Yeah...and we also should start thinking about splitting algorithms.hpp to put the C & C++ parts in sync.

Reply:
Unlike the previous PR where you defined the centrality result types in both C and pylib, here you're splitting up `algorithms.h` into itself and `centrality_algorithms.h`. I think it would be better and easier to follow along if I do the same in my pylibcugraph wrapper PR for `algorithms.pxd` and `centrality_algorithms.pxd`. It would likely prevent a merge conflict and I can keep pulling from your branch whenever changes are being made

Reply:
@betochimas - that makes sense.  Did you want me to split that in this PR, or are you going to handle it in #2201?

Reply:
@ChuckHastings I can handle it in #2201

Reply:
yep, I think @betochimas is saying this too: we've been keeping the .pxd files in `pylibcugraph/_cugraph_c` in sync with the C API headers to make it easy to follow, so we'd want the same reorg in pylibcugraph. I think it can be done in #2201.